base/src/main/java/org/robojackets/apiary/base/ui/nfc/BuzzCardPrompt.kt

```diff
@@ -96,30 +106,100 @@ fun BuzzCardPrompt(
                     else -> UnknownNfcError
                 }
             } catch (e: NumberFormatException) {
-                Log.e(tag, "GTID string from (probably a) BuzzCard could not be parsed as an Int", e)
+                Log.e(
+                    tag,
+                    "GTID string from (probably a) BuzzCard could not be parsed as an Int",
+                    e
+                )
                 error = InvalidBuzzCardData
             }
         },
         NfcAdapter.FLAG_READER_NFC_A // NFC adapter flags, BuzzCards are Type A according to TagInfo
     )
 
-    // Make sure we stop responded to card taps when this Composable is disposed (no longer on
+    // Make sure we stop responding to card taps when this Composable is disposed (no longer on
     // screen)
     DisposableEffect(Unit) {
         onDispose {
             nfcLib.disableReaderMode()
         }
     }
 
-    if (!hidePrompt) {
-        when (error) {
-            null -> BuzzCardReadyForTap()
-            TagLost -> NfcTagLostError()
-            NotABuzzCard -> NfcNotABuzzCardError()
-            InvalidBuzzCardData -> NfcInvalidBuzzCardDataError()
-            UnknownNfcError -> NfcReadUnknownError()
+    var showGtidPrompt by remember { mutableStateOf(false) }
+    Column {
+        if (!hidePrompt) {
+            when (error) {
+                null -> BuzzCardReadyForTap()
+                TagLost -> NfcTagLostError()
+                NotABuzzCard -> NfcNotABuzzCardError()
+                InvalidBuzzCardData -> NfcInvalidBuzzCardDataError()
+                UnknownNfcError -> NfcReadUnknownError()
+            }
+            Button(
+                onClick = { showGtidPrompt = true },
+                Modifier.align(CenterHorizontally)
+            ) {
+                Text("Enter GTID manually")
+            }
         }
     }
+
+    if (showGtidPrompt) {
+        ManualGtidEntryPrompt(
+            onGtidEntered = {
+                onBuzzCardTap(it)
+                error = null
+            },
+            onHide = { showGtidPrompt = false }
+        )
+    }
+}
+
+@Composable
+fun ManualGtidEntryPrompt(onHide: () -> Unit, onGtidEntered: (entry: BuzzCardTap) -> Unit) {
+    var gtid by remember { mutableStateOf("") }
+    AlertDialog(
+        onDismissRequest = {
+            gtid = ""
+            onHide()
+        },
+        confirmButton = {
+            TextButton(onClick = {
+                if (gtidRegex.matches(gtid)) {
+                    onGtidEntered(BuzzCardTap(gtid.toInt(), source = Keyboard))
+                    onHide()
+                    gtid = ""
+                }
+            }) {
+                Text("Submit")
+            }
+        },
+        title = { Text(text = "Manual GTID entry", style = MaterialTheme.typography.h5) },
+        text = {
+            Column {
+                Text("Type the entire GTID, starting with 9")
+                OutlinedTextField(
```

Comment:
Can you make the text field grab focus when the dialog appears? Right now I have to tap the text box to have the keyboard appear which adds a step.

Reply:
In theory, yes, but in practice I couldn't quite get it to work and don't think it's worth the trouble at the moment, considering this isn't intended to be a primary mode of use